### PR TITLE
Validation of input strings

### PR DESCRIPTION
**Pull request by rhn**
These patches add 2 things: 
- rejecting input strings which contain `CR` or `LF` characters in what was previously (#56) handled just by `quote!`
- passing on the actual response inside `BadResponse`

This pull request is based on #56

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -601,7 +622,7 @@ mod tests {
         let response = b"a1 OK Logged in\r\n".to_vec();
         let username = "username";
         let password = "password";
-        let command = format!("a1 LOGIN {} {}\r\n", username, password);
+		let command = format!("a1 LOGIN {} {}\r\n", quote!(username), quote!(password));
```
Comment (jonhoo) [inline src/client.rs]: Once this is fixed in #56, you'll want to rebase.

File: src/error.rs [hunk 2]
```diff
@@ -62,6 +64,9 @@ impl fmt::Display for Error {
             Error::Io(ref e) => fmt::Display::fmt(e, f),
             Error::Tls(ref e) => fmt::Display::fmt(e, f),
             Error::TlsHandshake(ref e) => fmt::Display::fmt(e, f),
+            Error::BadResponse(ref data) => {
```
Comment (jonhoo) [inline src/error.rs]: This seems like an unrelated change?
Also, shouldn't `Error::Validate` have an entry here?

Comment (rhn) [reply src/error.rs]: There are 2 changes in this set: expanding `BadResponse` and also adding validation. Should I split them into separate PRs?

`Error::Validate` added in `Display::fmt`.

**Comment by jonhoo** [comment]
LGTM, thanks!

**Comment by rhn** [comment]
Thanks!
